core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.ts

```diff
@@ -332,6 +344,9 @@ oppia.factory('ExplorationStatesService', [
       },
       saveSolution: function(stateName, newSolution) {
         saveStateProperty(stateName, 'solution', newSolution);
+        stateInteractionSavedCallbacks.forEach(function(callback) {
```

Comment:
I think I am a little confused by this. I wouldn't normally think of changes to the solution/hints as changes to the "interaction".

Also, why do we need this hook (how does it affect the top answers)?

Reply:
Ack, I won't call it for solutions and hints.

This hook recalculates the answers to check that they're still relevant. AFAICT, this needs to happen when:

- Unclassified answers changes (what if a student enters it anyway?)
- Interaction Answer Groups changes (some answers may no longer/now have an explicit answer)
- Interaction ID changes (our old answers aren't relevant anymore, would need to wait for new ones).

Reply:
Sg. Though, what do you mean by "what if a student enters it anyway?"

Reply:
Ah, the definition of an "unclassified answer" changes to something an answer previously entered (or no longer becomes one).